lib/src/main/kotlin/com/swmansion/starknet/deployercontract/StandardDeployer.kt

```diff
@@ -0,0 +1,60 @@
+package com.swmansion.starknet.deployercontract
+
+import com.swmansion.starknet.account.Account
+import com.swmansion.starknet.data.selectorFromName
+import com.swmansion.starknet.data.types.Call
+import com.swmansion.starknet.data.types.Calldata
+import com.swmansion.starknet.data.types.Event
+import com.swmansion.starknet.data.types.Felt
+import com.swmansion.starknet.data.types.transactions.GatewayTransactionReceipt
+import com.swmansion.starknet.data.types.transactions.InvokeTransactionReceipt
+import com.swmansion.starknet.data.types.transactions.TransactionReceipt
+import com.swmansion.starknet.extensions.map
+import com.swmansion.starknet.provider.Provider
+import com.swmansion.starknet.provider.Request
+
+class StandardDeployer(
+    private val deployerAddress: Felt,
+    private val provider: Provider,
+    private val account: Account,
+) : Deployer {
+    override fun deployContract(
```

Comment:
I would also offer a deployContract with some default parameters: unique = True and salt generated with bouncy castle.